src/tools/auth0/handlers/hooks.js

```diff
@@ -150,7 +150,7 @@ export default class HooksHandler extends DefaultHandler {
       // hooks.getAll does not return code and secrets, we have to fetch hooks one-by-one
       this.existing = await Promise.all(hooks.map((hook) => this.client.hooks.get({ id: hook.id })
         .then((hookWithCode) => this.client.hooks.getSecrets({ id: hook.id })
-          .then((secrets) => ({ ...hookWithCode, secrets })))));
+          .then((secrets) => ({ name: hook.id, ...hookWithCode, secrets })))));
```

Comment:
This LGTM, but should the proper fix be in API2 so that we always return a name on the hook result?

Reply:
SGTM. After peeling back 20 (unnecessary?) layers of abstraction, the issue seems to be here: https://github.com/auth0/auth0-hooks-api/blob/0f801814243dd6e8b620900f1d6b4b4f42f2441b/packages/hooks-api-core/src/hooks/mappers/HookMapper.ts#L9-L17

Reply:
https://github.com/auth0/auth0-hooks-api/pull/51